library/network_connections.py

```diff
@@ -428,6 +428,19 @@ def ifcfg_create(
                 connection["interface_name"],
                 " ".join(configured_coalesce),
             )
+        ethtool_ring = connection["ethtool"]["ring"]
+        configured_ring = []
+        for ring, setting in ethtool_ring.items():
+            if setting is not None:
```

Comment:
yes, right. But as I tried to say earlier, above is also
```python
         for coalesce, setting in ethtool_coalesce.items():
             if setting is not None:
                 if isinstance(setting, bool):
                     setting = int(setting)
                 configured_coalesce.append(
```
with the same problem. Please fix this pre-existing code in a separate commit.

Reply:
IMHO, I do not think we should remove  ` if isinstance(setting, bool): setting = int(setting)` for coalesce setting. Because the data type for `adaptive_rx` and `adaptive_tx` in Network role is bool, see
```python
                ArgValidatorBool("adaptive_rx", default_value=None),
                ArgValidatorBool("adaptive_tx", default_value=None),
```
But In NetworkManager, we have to set it to either 0 or 1 when modify the `ethtool.coalesce-adaptive-rx` and `ethtool.coalesce-adaptive-tx` setting.
```bash
[root@localhost ~]# nmcli connection modify testnic1 ethtool.coalesce-adaptive-rx True
Error: failed to modify ethtool.coalesce-adaptive-rx: 'True' is out of range [0, 4294967295].# Is anything wrong with the error msg?
[root@localhost ~]# nmcli connection modify testnic1 ethtool.coalesce-adaptive-rx 1
```